modules/5_tracking/tracker.py

```diff
@@ -133,13 +133,20 @@ def _calc_affinity(self, track, obj):
         return affinity_appearance * affinity_position * affinity_shape
 
     def _calc_affinity_appearance(self, track, obj):
-        raise NotImplementedError("The function _calc_affinity_appearance  is not implemented -- implement it by yourself")
+        return calc_features_similarity(track.last().appearance_feature,obj.appearance_feature)
 
     def _calc_affinity_position(self, track, obj):
-        raise NotImplementedError("The function _calc_affinity_position is not implemented -- implement it by yourself")
+        C1 = get_bbox_center(track.last().bbox)
+        C2 = get_bbox_center(obj.bbox)
+        dist = get_dist(C1,C2)
+        return math.exp(-0.5*dist**2 / calc_bbox_area(track.last().bbox)
 
     def _calc_affinity_shape(self, track, obj):
```

Comment:
I don't see a mistake here

Reply:
Please try to commit something to trigger CI

Reply:
There are three opening parentheses but only two closing ones